### ISSUE TITLE
directory import fails again since 4.15.8

### ISSUE THREAD

**Issue by TonyRL** [privatenumber/tsx#607]
### Acknowledgements

- [X] I read the documentation and searched existing issues to avoid duplicates
- [X] I understand this is a **bug tracker** and anything other than a proven bug will be closed
- [X] I understand this is a free project and relies on community contributions
- [X] I read and understood the [Contribution guide](https://github.com/privatenumber/tsx/blob/master/CONTRIBUTING.md)

### Minimal reproduction URL

https://github.com/TonyRL/tsx-4.15.8-repro

### Problem & expected behavior (under 200 words)

Similar to #579 and #583, but with `"type": "module"`.

- What happened (in 4.15.8 ~ 4.16.0)
```sh
$ pnpm run start
> tsx-repro@1.0.0 start /workspace/tsx-repro
> tsx index.ts


node:internal/modules/run_main:115
    triggerUncaughtException(
    ^
Error [ERR_MODULE_NOT_FOUND]: Cannot find package '@/hello' imported from /workspace/tsx-repro/index.ts
    at packageResolve (node:internal/modules/esm/resolve:841:9)
    at moduleResolve (node:internal/modules/esm/resolve:914:18)
    at defaultResolve (node:internal/modules/esm/resolve:1119:11)
    at nextResolve (node:internal/modules/esm/hooks:791:28)
    at resolveBase (file:///workspace/tsx-repro/node_modules/.pnpm/tsx@4.16.0/node_modules/tsx/dist/esm/index.mjs?1719848400741:2:3233)
    at async resolveDirectory (file:///workspace/tsx-repro/node_modules/.pnpm/tsx@4.16.0/node_modules/tsx/dist/esm/index.mjs?1719848400741:2:3519)
    at async resolve (file:///workspace/tsx-repro/node_modules/.pnpm/tsx@4.16.0/node_modules/tsx/dist/esm/index.mjs?1719848400741:2:4036)
    at async nextResolve (node:internal/modules/esm/hooks:791:22)
    at async Hooks.resolve (node:internal/modules/esm/hooks:238:24)
    at async handleMessage (node:internal/modules/esm/worker:255:18) {
  code: 'ERR_MODULE_NOT_FOUND'
}

Node.js v22.2.0
```

- What I expected (in 4.15.7)
```sh
$ pnpm run start
> tsx-repro@1.0.0 start /workspace/tsx-repro
> tsx index.ts

Hello, world!
```

### Bugs are expected to be fixed by those affected by it

- [ ] I'm interested in working on this issue

### Compensating engineering work will speed up resolution and support the project

- [ ] I'm willing to offer $10 for financial support

**Comment by shinebayar-g** [comment]
Yup tsx is broken since 4.15.8

**Comment by rohit-ks** [comment]
+1

**Comment by Infinitay** [comment]
I thought I was the only one having this issue. I made a repo too with the problem that I'm having: https://github.com/Infinitay/path-alias-issue

Although the odd thing is that if I import a type, it works fine - the issue is when I import an enum and reference/access it.

**Comment by privatenumber** [comment]
As per the [Contribution Guide](https://github.com/privatenumber/tsx/blob/master/CONTRIBUTING.md#-avoid-detractive-comments), please refrain from "me too" comments.

I haven't taken a look at this yet but if anyone is interested in contributing constructively, feel free to open a PR.

**Comment by privatenumber** [comment]
:tada: This issue has been resolved in v4.16.1

If you rely on this project, please consider [making a donation](https://github.com/sponsors/privatenumber) :heart: :pray:
